TileFormats/BatchTable/README.md

```diff
@@ -147,6 +164,344 @@ var cartographicArray = new Float64Array(batchTableBinary.buffer, byteOffset, ca
+carType : "sedan"
+carColor : "red"
+```
+
+**Figure 3**: Batch Table Hierarchy - Parking Lot
+
+![batch table hierarchy parking lot](figures/batch-table-hierarchy-parking-lot.png)
+
+#### Feature Hierarchy
+
+The city block example would now look like:
+
+```json
+{
+  "HIERARCHY" : {
+    "classes" : [
+      {
+        "name" : "Wall",
+        "length" : 6,
+        "instances" : {
+          "wall_color" : ["blue", "pink", "green", "lime", "black", "brown"],
+          "wall_windows" : [2, 4, 4, 2, 0, 3]
+        }
+      },
+      {
+        "name" : "Building",
+        "length" : 3,
+        "instances" : {
+          "building_name" : ["building_0", "building_1", "building_2"],
+          "building_id" : [0, 1, 2],
+          "building_address" : ["10 Main St", "12 Main St", "14 Main St"]
+        }
+      },
+      {
+        "name" : "Block",
+        "length" : 1,
+        "instances" : {
+          "block_lat_long" : [[0.12, 0.543]],
+          "block_district" : ["central"]
+        }
+      }
+    ],
+    "instancesLength" : 10,
+    "classIds" : [0, 0, 0, 0, 0, 0, 1, 1, 1, 2],
+    "parentIds" : [6, 6, 7, 7, 8, 8, 9, 9, 9, 9]
+  }
+}
+```
+
+The tile's `batchLength` is 6 and `instancesLength` is 10. The building and block instances are not features of the tile, but contain properties that are inherited by the six wall features.
+
+`parentCounts` is not included since every instance has at most one parent.
+
+A feature with `batchId = 3` has the following properties:
+
+```
+wall_color : "lime"
+wall_windows : 2
+building_name : "building_1"
+building_id : 1,
+building_address : "12 Main St"
+block_lat_long : [0.12, 0.543]
+block_district : "central"
+```
+
+Breaking it down into smaller steps:
+
+The feature with `batchId = 3` is the fourth "Wall" instance, and its properties are:
+```
+wall_color : "lime"
+wall_windows : 2
+```
+
+The feature's `parentId` is 7, which is the second "Building" instance. Therefore it gets the following properties from its parent:
+```
+building_name : "building_1"
+building_id : 1,
+building_address : "12 Main St"
+```
+
+The building's `parentId` is 9, which is the sole "Block" instance with the following properties:
+```
+block_lat_long : [[0.12, 0.543]]
+block_district : ["central"]
+```
+
+Since the block's `parentId` is also 9, it does not have a parent and the traversal is complete.
+
+**Figure 4**: Batch Table Hierarchy - Block
+
+![batch table hierarchy block](figures/batch-table-hierarchy-block.png)
+
+### Styling
+
+The styling language supports additional functions for querying feature classes:
+
+* `isExactClass`
+* `isClass`
+* `getExactClassName`
+
+More detailed descriptions are provided in the [Styling Spec](../../Styling/README.md#batch-table-hierarchy).
+
+### Notes
+
+* In some cases a feature may have multiple ancestors of the same class, or ancestors of different classes with the same property names. It is up to the implemantation to decide how to handle overloaded properties.
```

Comment:
Can you provide a concrete example here.

> It is up to the implemantation to decide how to handle overloaded properties.

In general, we need to avoid undefined behavior as much as possible.  Are we sure we can't define reasonably behavior here?